### PR TITLE
[improve](node) refactor partition sort node to reduce memory use

### PR DESCRIPTION
**Pull request by zhangstar333**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

File: be/src/vec/exec/vpartition_sort_node.cpp [hunk 1]
```diff
@@ -33,10 +33,50 @@
 #include "vec/common/hash_table/hash_map_context_creator.h"
 #include "vec/common/hash_table/hash_set.h"
 #include "vec/common/hash_table/partitioned_hash_map.h"
+#include "vec/core/block.h"
 #include "vec/exprs/vexpr.h"
 #include "vec/exprs/vexpr_context.h"
 
 namespace doris::vectorized {
+Status PartitionBlocks::do_partition_topn_sort() {
+    if (_partition_topn_sorter == nullptr) {
```
Comment (HappenLee) [inline be/src/vec/exec/vpartition_sort_node.cpp:42]: why each time need create the new `_partition_topn_sorter`

File: be/src/vec/exec/vpartition_sort_node.cpp [hunk 2]
```diff
@@ -79,13 +119,18 @@ Status VPartitionSortNode::prepare(RuntimeState* state) {
...
     _emplace_key_timer = ADD_TIMER(runtime_profile(), "EmplaceKeyTime");
+    runtime_profile()->add_info_string("CurrentTopNPhase", std::to_string(_topn_phase));
 
     RETURN_IF_ERROR(ExecNode::prepare(state));
     SCOPED_TIMER(_exec_timer);
     RETURN_IF_ERROR(_vsort_exec_exprs.prepare(state, child(0)->row_desc(), _row_descriptor));
     RETURN_IF_ERROR(VExpr::prepare(_partition_expr_ctxs, state, child(0)->row_desc()));
     _init_hash_method();
 
+    _partition_sort_info = std::make_shared<PartitionSortInfo>(
```
Comment (HappenLee) [inline be/src/vec/exec/vpartition_sort_node.cpp:130]: why use shared_ptr？not unique_ptr?

**Review by yiguolei — APPROVED** [review]

**Review by mrhhsg — APPROVED** [review]
LGTM

**Pull request merged**